brave-okhttp/src/main/java/com/github/kristofa/brave/okhttp/OkHttpRequest.java

```diff
@@ -0,0 +1,36 @@
+package com.github.kristofa.brave.okhttp;
+
+
+import com.github.kristofa.brave.http.HttpClientRequest;
+import okhttp3.Request;
+
+import java.net.URI;
+
+class OkHttpRequest implements HttpClientRequest {
+
+    private final Request.Builder requestBuilder;
+
+    OkHttpRequest(Request.Builder requestBuilder) {
+        this.requestBuilder = requestBuilder;
+    }
+
+    @Override
+    public void addHeader(String header, String value) {
+        requestBuilder.addHeader(header, value);
+    }
+
+    @Override
+    public String getHttpMethod() {
+        return getRequest().method();
+    }
+
+    @Override
+    public URI getUri() {
+        return getRequest().url().uri();
```

Comment:
It's wasteful to be calling `getRequest()` in these methods. Consider taking in the original `Request` and storing it in a field alongside the mutable `newBuilder()`